jdaviz/configs/cubeviz/cubeviz.ipynb

```diff
@@ -6,10 +6,11 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "import pathlib\n",
     "from jdaviz.app import Application\n",
     "from glue.core import Data\n",
     "app = Application(configuration='cubeviz')\n",
-    "data = app.load_data('DATA_FILENAME')\n",
+    "data = app.load_data(str(pathlib.Path('DATA_FILENAME').absolute()))\n",
```

Comment:
Agreed with @nmearl, but it raises the question: should `load_data` have a:
```
datapath = str(datapath)
```
at the top? I can imagine users wanting to pass around `pathlib` paths and `load_data`, etc should be able to handle it

Reply:
Agreed with @nmearl, but it raises the question: should `load_data` have a:
```
datapath = str(datapath)
```
at the top? I can imagine users wanting to pass around `pathlib` paths and `load_data`, etc should be able to handle it

Reply:
Gotcha, that sounds good to me!

Reply:
> I added it in both places just in case someone tries to sideload the notebook rather than going through our CLI. I generally err on the side of being overly cautious, but I can also see the logic of just assuming the filepath is valid.

Right, but users who as in a windows environment are just going to type in their normal file path. It seems weird to have to specify specifically for users that want to use the notebook directly when in their normal operations, this wouldn't be necessary.

> To give a quick sense of how this line helped me specifically is that when the main() method printed out my paths, on windows, paths need double backslashes ("C:\\"). This was rendered in the notebook as "C:\" when it was printed out, so when it was loaded back in by voila, it failed.

I'm confused to why it would give a different path? Wouldn't having 

```python
nbf.write(notebook_template.replace('DATA_FILENAME', str(pathlib.Path(filepath).absolute()).encode('unicode_escape').decode()).strip())
```

produce the same thing?

Reply:
For the first part, I'd argue doing a `pathlib.Path.absolute` enforces standardizing the filepath downstream such that the app and parsers don't have to think about properly resolving the path. I.e. if being run through the CLI (with the line you suggest above), then you get the full resolved path written to the notebook. But I could also see a user going in and sideloading the notebook and only putting in a relative path to their file. Then, it's on the onus of `app.load_data` do to the checking and path resolution (or in reality the parsers further down the line). I'd say let's not even have the parsers have to think about it and force standardization as far up the chain as possible. That way the scope of potential different variations of arguments for `app.load_data` at least decreases a little. I could definitely see eliminating the `absolute` call in `cli.main` and reserving it for inside the config notebooks?

As for the second point, I see what you're going for. My apologies, I misunderstood. The secret sauce is the `encode('unicode_escape')` and corresponding `decode` call for @eteq's comment.

Reply:
> For the first part, I'd argue doing a pathlib.Path.absolute enforces standardizing the filepath downstream such that the app and parsers don't have to think about properly resolving the path

Sure, but this comes at the expense of placing another barrier in front of the users, who would then need to understand the behavior and use of `pathlib` instead of simply knowing "I need to write my file path here".

> Then, it's on the onus of app.load_data do to the checking and path resolution (or in reality the parsers further down the line).

I'm not quite sure the argument here: it's really pretty straightforward to parse a file path. If we need to convert the relative path to an absolute one we should certainly do that; `pathlib` et al. are more than capable of making that easy for us. I also do think it's our responsibility, and not the user's, to ensure that their valid path is correctly understand and dealt with in the internals of jdaviz.

> I'd say let's not even have the parsers have to think about it and force standardization as far up the chain as possible.

I agree! But I stop at the moment this becomes exposed to the user.